src/Microsoft.AspNet.Mvc.Core/ModelBinding/CollectionModelBinder.cs

```diff
@@ -149,12 +149,16 @@ protected object CreateInstance(Type targetType)
                 bindingContext.ModelName,
                 bindingContext.ModelMetadata,
                 boundCollection);
+
+            var innerBindingContext = ModelBindingContext.CreateChildBindingContext(
+                bindingContext,
+                elementMetadata,
+                fieldName: string.Empty,
```

Comment:
Suggest `bindingContext.FieldName` since inner binder is handling (individual values for) the same property.